packages/core/contracts/trader/uniswap-broker/UniswapBroker.sol

```diff
@@ -0,0 +1,171 @@
+        }
+
+        require(amountIn > 0, "SwapToPrice: ZERO_AMOUNT_IN");
+
+        // spend up to the allowance of the token in
+        uint256 maxSpend = aToB ? maxSpendTokenA : maxSpendTokenB;
+        if (amountIn > maxSpend) {
+            amountIn = maxSpend;
+        }
+
+        address tokenIn = aToB ? tokenA : tokenB;
+        address tokenOut = aToB ? tokenB : tokenA;
+        TransferHelper.safeTransferFrom(tokenIn, msg.sender, address(this), amountIn);
+        TransferHelper.safeApprove(tokenIn, address(router), amountIn);
+
+        address[] memory path = new address[](2);
+        path[0] = tokenIn;
+        path[1] = tokenOut;
+
+        router.swapExactTokensForTokens(
+            amountIn,
+            0, // amountOutMin: we can skip computing this number because the math is tested
+            path,
+            to,
+            deadline
+        );
+    }
+
+/**
+     * @notice Given the "true" price a token (represented by truePriceTokenA/truePriceTokenB) and the reservers in the
+     * uniswap pair, calculate: a) the direction of trade (aToB) and b) the amount needed to trade (amountIn) to move
+     * the pool price to be equalto the true price.
+     * @dev Note that this method uses the Babylonian square root method which has a small margin of error which will
+     * result in a small over or under estimation on the size of the trade needed.
+     * @param truePriceTokenA the nominator of the true price.
+     * @param truePriceTokenB the denominatornominator of the true price.
+     * @param reserveA number of token A in the pair reserves
+     * @param reserveB number of token B in the pair reserves
+     */
+//
+    function computeTradeToMoveMarket(
+        uint256 truePriceTokenA,
+        uint256 truePriceTokenB,
+        uint256 reserveA,
+        uint256 reserveB
+    ) public pure returns (bool aToB, uint256 amountIn) {
+        aToB = FullMath.mulDiv(reserveA, truePriceTokenB, reserveB) < truePriceTokenA;
+
+        uint256 invariant = reserveA.mul(reserveB);
+
+        uint256 leftSide =
+            Babylonian.sqrt(
+                FullMath.mulDiv(
+                    invariant,
+                    aToB ? truePriceTokenA : truePriceTokenB,
+                    aToB ? truePriceTokenB : truePriceTokenA
+                )
+            );
+        uint256 rightSide = (aToB ? reserveA : reserveB);
+
+        if (leftSide < rightSide) return (false, 0);
+
+        // compute the amount that must be sent to move the price back to the true price.
+        amountIn = leftSide.sub(rightSide);
+    }
+
+    // The methods below are taken from https://github.com/Uniswap/uniswap-v2-periphery/blob/master/contracts/libraries/UniswapV2Library.sol
+    // We could import this library into this contract but this library is dependent Uniswap's SafeMath, which is bound
+    // to solidity 6.6.6. Hardhat can easily deal with two diffrent sets of solidity versions within one project so 
```

Comment:
Another good reason to drop truffle support entirely!

Reply:
yup! however we currently have other packages that still run truffle tests so we still need to be able to run truffle compile :)